File: 0011-container-with-most-water/0011-container-with-most-water.cpp
class Solution {
public:
    int maxArea(vector<int>& height) {
        int n=height.size();
       int mini=0,maxi=INT_MIN,a=0;
        int i=0,j=n-1;
        while(i<j){
            mini=min(height[i],height[j]);
            a=j-i;
            maxi=max(maxi,a*mini);
            a=0;
            if(height[i]<height[j]){
                i++;
            }
            else{j--;}
        }
        return maxi;
    }
};

File: 0011-container-with-most-water/0011-container-with-most-water.java
class Solution {
    public int maxArea(int[] height) {
        int maxi = Integer.MIN_VALUE;
        int n = height.length;
        int i=0;int j=n-1;
        while(i<=j){
            int mini = Math.min(height[i], height[j]);
            int ar = j-i;
            maxi = Math.max(ar * mini , maxi);
            if(height[i] <= height[j]){
                i++;
            }
            else{
                j--;
            }
        }
        return maxi;
    }
}